<!--
 Copyright (c) 2023  Carnegie Mellon University, IBM Corporation, and others

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
-->

<launch>
  <!-- mapping args -->
  <arg name="output" default="log"/>
  <arg name="use_sim_time" default="false" />
  <arg name="configuration_basename" default="cabot.lua"/>
  <arg name="points2_topic" default="velodyne_points"/>
  <arg name="imu_topic" default="imu/data"/>
  <arg name="bagfile" default=""/>

  <!-- set use_sim_time flag -->
  <param name="/use_sim_time" value="$(arg use_sim_time)"/>

  <!-- cartographer_node -->
  <node pkg="cabot_explore" type="cartographer_explore.py" name="cartographer_explore" output="$(arg output)">
    <param name="configuration_directory" value="$(find cabot_explore)/configuration_files"/>
    <param name="configuration_basename" value="$(arg configuration_basename)"/>
    <param name="points2_topic" value="$(arg points2_topic)"/>
    <param name="imu_topic" value="$(arg imu_topic)"/>
  </node>

  <!-- cartographer_occupancy_grid_node -->
  <node pkg="cartographer_ros" type="cartographer_occupancy_grid_node"
        name="cartographer_occupancy_grid_node"
        args="-resolution 0.05 -publish_period_sec 0.5" />

  <node pkg="cabot_explore" type="record_data.py" name="record_data">
    <param name="bagfile" value="$(arg bagfile)"/>
  </node>
</launch>
